lib/prompt/parent_branches.go

```diff
@@ -33,5 +46,70 @@ func askForBranchAncestry(branchName string) {
 }
 
 func askForParentBranch(branchName string) string {
-	panic("unimplemented")
+	branchNameFmt := color.New(color.Bold).Add(color.FgCyan)
+	mainBranch := config.GetMainBranch()
+	message := fmt.Sprintf("Please specify the parent branch of %s by name or number (default: %s): ", branchNameFmt.Sprintf(branchName), mainBranch)
+	numberedBranches := git.GetLocalBranchesWithMainBranchFirst()
+	numericRegex, err := regexp.Compile("^[0-9]+$")
+	if err != nil {
+		log.Fatal("Error compiling numeric regular expression: ", err)
+	}
+
+	for {
+		fmt.Printf(message)
+		userInput := util.GetUserInput()
+		parent := ""
+		if numericRegex.MatchString(userInput) {
```

Comment:
Yes, but now we no longer have the excuse that we don't have a "real programming language"  :)

It's also interesting to see whether we can meaningfully simplify this, or if our rule of "methods should not have more than 10 loc" has limitations like this one.